### PR TITLE
Reuse compute budget processing

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Compute budget instruction processing for transactions is performed multiple times in the transaction pipeline. The computed budget limits should be reused through the pipeline.

#### Summary of Changes
- Remove one instance of re-processing compute budget instructions in the SVM transaction processor 
- Add metrics for transactions rejected due to invalid compute budget issues (this should never happen since banking stage filters these transactions out)
- Fail earlier in SVM if compute budget processing fails (again, this shouldn't make any difference because banking stage filters these transactions)

Fixes #

### CODE REVIEW

File: svm/src/transaction_processor.rs [hunk 1]
```diff
@@ -462,7 +445,15 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
...
         error_counters: &mut TransactionErrorMetrics,
-    ) -> transaction::Result<(FeeDetails, AccountSharedData, u64)> {
+    ) -> transaction::Result<(ComputeBudgetLimits, FeeDetails, AccountSharedData, u64)> {
+        let compute_budget_limits = process_compute_budget_instructions(
+            message.program_instructions_iter(),
+        )
+        .map_err(|err| {
+            error_counters.invalid_compute_budget += 1;
+            err
+        })?;
```
Comment (jstarry) [inline svm/src/transaction_processor.rs:425]: note that we now fail with an error immediately here in `validate_transaction_fee_payer` if compute budget processing fails rather than waiting until execution to fail. Both before and after this refactoring we mark such a transaction as invalid and don't execute it so this won't cause consensus issues.

**Review by tao-stones — COMMENTED** [review]
looks good overall, it removes the duplicated calling of `process_compute_budget_instructions` in "load_and_execute" path. I'll do another round of review in detail.

File: core/src/banking_stage/consume_worker.rs [hunk 2]
```diff
@@ -644,6 +649,12 @@ impl ConsumeWorkerTransactionErrorMetrics {
                     .swap(0, Ordering::Relaxed),
                 i64
             ),
+            (
+                "invalid_compute_budget",
+                self.invalid_compute_budget
+                    .swap(0, Ordering::Relaxed),
+                i64
+            ),
```
Comment (tao-stones) [inline core/src/banking_stage/consume_worker.rs:657]: Probably doesn't belong to this PR, just wondering if this error count is reported from replay as well?

File: svm/src/transaction_processor.rs [hunk 3]
```diff
@@ -299,31 +299,9 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
...
-                            Measure::start("compute_budget_process_transaction_time");
-                        let maybe_compute_budget = ComputeBudget::try_from_instructions(
-                            tx.message().program_instructions_iter(),
-                        );
-                        compute_budget_process_transaction_time.stop();
-                        saturating_add_assign!(
-                            execute_timings
-                                .execute_accessories
-                                .compute_budget_process_transaction_us,
-                            compute_budget_process_transaction_time.as_us()
```
Comment (tao-stones) [inline svm/src/transaction_processor.rs:314]: nit: Personally i don't really pay much attention to this metric, cause they are pretty much flat all the time, so removing it is fine with me. But in case there are queries read it, you can move it to `validate_transaction_fee_payer()` to keep "backward compatible". Totally up to you.

File: svm/src/transaction_processor.rs [hunk 4]
```diff
@@ -742,6 +736,10 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         let lamports_before_tx =
             transaction_accounts_lamports_sum(&transaction_accounts, tx.message()).unwrap_or(0);
 
+        let compute_budget = config
+            .compute_budget
```
Comment (tao-stones) [inline svm/src/transaction_processor.rs:727]: +1 for keeping config overrides.

**Review by tao-stones — DISMISSED** [review]
lgtm, a small nit about metric change shouldn't not block this PR being merged.

File: core/src/banking_stage/consume_worker.rs (hunk 2 above)
Comment (jstarry) [reply core/src/banking_stage/consume_worker.rs:657]: Nope, only from banking stage. I don't think it makes too much sense to report stats for individual error types from replay because it would mean the block is invalid and that rarely if ever happens.

File: svm/src/transaction_processor.rs (hunk 3 above)
Comment (jstarry) [reply svm/src/transaction_processor.rs:314]: I'd prefer to remove it, the compute budget processing is now tracked under `"validate_fees_us"` as of https://github.com/anza-xyz/agave/pull/1529

**Comment by jstarry** [comment]
@tao-stones, I just rebased to fix conflicts, can you take another look?

**Review by tao-stones — APPROVED** [review]
lgtm
